src/commands/basemaps-mapsheet/create-mapsheet.ts

```diff
@@ -39,17 +41,17 @@ interface Output {
 export const CommandCreateMapSheetArgs = {
   verbose,
   path: option({
-    type: string,
+    type: PathStringOrUrlStringFromString,
```

Comment:
This should be URL, anytime we are taking user input we should be validating that it does actually meet the standard of being a URL.

We use three common URL parsers in basemaps which could be replicated here:

https://github.com/linz/basemaps/blob/master/packages/cogify/src/cogify/parsers.ts

Reply:
Do we actually need `UrlFolder` or `UrlArrayJsonFile` anywhere in argo-tasks?

Reply:
UrlFolder is very helpful to ensure that something is a folder and not a prefix

`s3://foo/bar` (prefix) vs `s3://foo/bar/` (folder) as when you join urls it has big implications as to where the file goes

for example adding `new URL("index.html", source)` to both of those would result in `s3://foo/index.html` and `s3://foo/bar/index.html`


UrlArrayJson is mostly  handled with the manifest files I think in this repo.

Reply:
I'll leave them out as long as we don't need them, though.